addons/sale_crm/models/sale_order.py

```diff
@@ -13,3 +13,23 @@ class SaleOrder(models.Model):
 
     def action_confirm(self):
         return super(SaleOrder, self.with_context({k:v for k,v in self._context.items() if k != 'default_tag_ids'})).action_confirm()
+
+    def write(self, vals):
+        res = super().write(vals)
+        if vals.get('state'):
+            self.opportunity_id.with_context(tracking_disable=False)._update_revenues_from_so()
+        return res
+
+    def _filter_sales_order(self, aggregation_field):
+        domain = [
+            ('opportunity_id', '=', self.opportunity_id.id),
+            ('state', '!=', 'cancel'),
+            ('currency_id', '=', self.opportunity_id.company_id.currency_id.id),
```

Comment:
I'm guessing this is why you do not use read_group on all opportunities at the same time?
Because we still need to only take the orders with relevant currencies?

Reply:
(Just writing my thoughts : Maybe we could remove that part of the domain, use the read_group to return recordsets of orders per opportunity that we can filter regarding the currency, then take the min amount. But then we would still be filtering and looping on every opportunity + SO.) We'll see what TDE thinks.

Reply:
Continuing my idea: But having a read_group called on a single opportunity each time, and grouping by opportunity, does not really make sense to me ^^